akka-docs/rst/java/persistence.rst

```diff
@@ -775,6 +775,29 @@ A snapshot of state data can be persisted by calling the ``saveStateSnapshot()``
 On recovery state data is initialized according to the latest available snapshot, then the remaining domain events are replayed, triggering the
 ``applyEvent`` method.
 
+
+Periodical snapshot
+-------------------
+
+You can enable periodical ``saveStateSnapshot()`` calls in ``PersistentFSM`` if you turn the following flag on in ``reference.conf``::
+
+    akka.persistence.fsm.enable-snapshot-after = on
+
+and also configure the ``snapshot-after`` value. For example if you specify::
+
+    akka.persistence.fsm.snapshot-after = 1000
+
+this means ``saveStateSnapshot()`` is called when the sequence number is multiple of 1000.
+
+.. note::
+
+ When you are calling ``applying`` method in the ``PersistentFSM`` DSL by passing multiple domain events,
+ then ``saveStateSnapshot()`` might not be called exactly at sequence numbers being multiple of the
+ ``snapshot-after`` configuration value.
+ This is because when ``applying`` takes multiple domain events, they are handled in a sort of "batch"
+ mode and  ``saveStateSnapshot()`` is called only at the end of batch.
```

Comment:
Isn't there one event for the state transition and one (or more) domain events? Could that result in that you accidentally get into a series of sequence numbers that never match the modulo condition?
E.g. always odd numbers will never trigger snapshots after 1000.
Could we handle that somehow?